src/Components/AppCompareTable/AppCompareTable.tsx

```diff
@@ -0,0 +1,121 @@
+import {
+  Box,
+  Paper,
+  Table,
+  TableBody,
+  TableContainer,
+  TableHead,
+  TableRow,
+} from "@mui/material";
+import React, { useEffect } from "react";
+import AppHistoryBlock from "../AppHistoryBlock/AppHistoryBlock";
+import { useAppSelector } from "../../redux/hooks";
+import AppGraphImage from "../AppGraphImage/AppGraphImage";
+import { styled } from "@mui/material/styles";
+import TableCell, { tableCellClasses } from "@mui/material/TableCell";
+import { Tiles } from "../../constants/tiles";
+import { usePostPathFindingCompareMutation } from "../../redux/rtk/pathFinding";
+import { formatMillisecondsToString } from "../../utils/utils";
+import { COLORS } from "../../constants/colors";
+
+const StyledTableCell = styled(TableCell)(({ theme }) => ({
+  [`&.${tableCellClasses.head}`]: {
+    backgroundColor: COLORS.COMPARE_TABLE_BGC,
+    color: COLORS.COMPARE_TABLE_COLOR,
+  },
+  [`&.${tableCellClasses.body}`]: {
+    fontSize: 14,
+  },
+}));
+
+function createData(name: any, data: any[]) {
+  console.log(data);
+  return { name, data };
+}
+
+const StyledTableRow = styled(TableRow)(({ theme }) => ({
+  "&:nth-of-type(odd)": {
+    backgroundColor: theme.palette.action.hover,
+  },
+  // hide last border
+  "&:last-child td, &:last-child th": {
+    border: 0,
+  },
+}));
+
+const AppCompareTable = () => {
+  // const [_, { data }] = usePostPathFindingCompareMutation();
+  const solution = useAppSelector((state) => state.pathFinding.solution);
+
+  const bfsTime = formatMillisecondsToString(solution?.bfs.time ?? 0);
+  const dfsTime = formatMillisecondsToString(solution?.dfs.time ?? 0);
+  const bidirectionalTime = formatMillisecondsToString(
+    solution?.bidirectional_search.time ?? 0
+  );
+  const aStarTime = formatMillisecondsToString(
+    solution?.a_star_search.time ?? 0
+  );
+  const rows = [
+    createData("Solution Graph", [
+      <AppGraphImage />,
+      <AppGraphImage />,
+      <AppGraphImage />,
+      <AppGraphImage />,
+    ]),
+    createData("Path Size", [
+      solution?.bfs.path.length,
+      solution?.dfs.path.length,
+      solution?.bidirectional_search.path.length,
+      solution?.a_star_search.path.length,
+    ]),
+    createData("Number Of Visited Tiles", [
+      solution?.bfs.visitedList.length,
+      solution?.dfs.visitedList.length,
+      solution?.bidirectional_search.visitedList.length,
+      solution?.a_star_search.visitedList.length,
+    ]),
+    createData("Time", [bfsTime, dfsTime, bidirectionalTime, aStarTime]),
```

Comment:
I really don't want you to code like this, it is ugly and not scalable. You should add an optional parameter to createData that is a transformation function that will be applied to all inputs, so what you should de looks like this:
 createData("Time", [solution?.bfs.time, solution?.dfs.time, solution?.bidirectional_search.time, solution?.a_star_search.time]), formatMillisecondsToString) and formatMillisecondsToString should be applied to all inputs in the array.